src/reason-parser/reason_oprint.ml

```diff
@@ -294,33 +294,39 @@ and print_arg ppf (lab, typ) =
   print_out_type_2 ppf typ;
   pp_print_string ppf suffix;
 
-and print_out_type_1 ppf =
+and print_out_type_1 ~uncurried ppf =
   function
     Otyp_arrow (lab, ty1, ty2) ->
       let rec collect_args args typ = match typ with
         | Otyp_arrow (lab, ty1, ty2) -> collect_args (args @ [(lab, ty1)]) ty2
         | _ -> (args, typ)
       in
+      let print_bs_uncurried_dot () =
+        if uncurried then fprintf ppf ".@ " else ()
+      in
       pp_open_box ppf 0;
       let (args, result) = collect_args [(lab, ty1)] ty2 in begin
-        match args with
-        (* single argument should not be wrapped... *)
-        | [(_, Otyp_tuple _) as arg] ->
+        match (uncurried, args) with
+        (* single argument should not be wrapped *)
+        (* though uncurried type are always wrapped in parens. `. a => 1` isn't supported *)
+        | (false, [(_, Otyp_tuple _) as arg]) ->
           (* ...unless it's for tuple. ((int, int)) => string *)
           pp_print_string ppf "(";
+          print_bs_uncurried_dot ();
```

Comment:
Remove? This line never prints.

Nit: the logic might be simpler by doing the wrapping in one place, and do the case analysis just to compute a boolan `should_wrap`.